src/components/content-tab-assistant.tsx

```diff
@@ -284,13 +310,27 @@ export function ContentTabAssistant( { selectedSite }: ContentTabAssistantProps
 	const isOffline = useOffline();
 	const { __ } = useI18n();
 	const lastMessage = messages.length === 0 ? undefined : messages[ messages.length - 1 ];
+	const hasFailedMessage = messages.some( ( msg ) => msg.failedMessage );
 
-	const handleSend = async ( messageToSend?: string ) => {
+	const handleSend = async ( messageToSend?: string, isRetry?: boolean ) => {
 		const chatMessage = messageToSend || input;
 		let messageId;
 		if ( chatMessage.trim() ) {
-			messageId = addMessage( chatMessage, 'user', chatId ); // Get the new message ID
-			setInput( '' );
+			if ( isRetry ) {
+				// If retrying, find the message ID with failedMessage flag
+				const failedMessage = messages.find(
+					( msg ) => msg.failedMessage && msg.content === chatMessage
+				);
```

Comment:
[Question/Proposal]: Now that we don't allow messages to be sent if one fails, isn't safe to assume that the failed message is always the last one, and thus, remove this find, @katinthehatsite ?

Reply:
I thought about it before but I prefer to keep searching for a specific message rather than relying on the last one. My reasoning for it is that if the logic for message handling changes in the future, the code will not break as it will still correctly identify the failed message regardless of its position.

Reply:
> I thought about it before but I prefer to keep searching for a specific message rather than relying on the last one.

For reference, the current logic finds the first error message, which doesn't exactly match your approach of retrying the last one. Not a blocker since this will always be the last one with the current implementation, but I wanted to highlight it in case this is an issue in the future.